pkg/operator/apiserver/controller/workload/workload.go

```diff
@@ -209,15 +210,23 @@ func (c *Controller) updateOperatorStatus(ctx context.Context, previousStatus *o
 
 		// we are degraded, not available and we are not progressing
 
-		deploymentDegradedCondition.Status = operatorv1.ConditionTrue
-		deploymentDegradedCondition.Reason = "PreconditionNotFulfilled"
-		deploymentDegradedCondition.Message = message
+		deploymentDegradedCondition = deploymentDegradedCondition.
+			WithStatus(operatorv1.ConditionTrue).
```

Comment:
maybe we should set status for all conditions in this case (`PreconditionNotFulfilled`) to `Unknown` ?

Reply:
Replied in the other comment.